netlogo-core/src/main/fileformat/PlotConverter.scala

```diff
@@ -100,30 +116,36 @@ class PlotConverter(
 
       override def apply(model: Model, modelPath: Path): ConversionResult = {
         val plotRenames = determineRenames(allPlotNames(model))
-        val penRenames = determineRenames(allPenNames(model))
-        if (plotRenames.isEmpty && penRenames.isEmpty) {
-          super.apply(model, modelPath)
-        } else {
-          val conversion = super.apply(model, modelPath)
-          conversion.updateModel(convertPlotAndPenNames(conversion.model, plotRenames.toMap, penRenames.toMap))
+        val penRenames: Seq[(String, Seq[(String,String)])] = determineMapRenames(allKeyedPenNames(model))
+        (plotRenames, penRenames) match {
+          case (Seq(), Seq()) => super.apply(model,modelPath)
+          case _ => { val conversion = super.apply(model, modelPath)
+            val convertPandPN: Model = convertPlotAndPenNames(conversion.model, plotRenames.toMap, penRenames.toMap)
+            conversion.updateModel(convertPandPN)
+          }
         }
       }
 
-      private def convertPlotAndPenNames(
-        model:       Model,
-        plotRenames: Map[String, String],
-        penRenames:  Map[String, String]): Model = {
-          val updatedWidgets = model.widgets.map {
-            case p: Plot =>
-              val newDisplay = p.display.map(currentName => plotRenames.getOrElse(currentName, currentName))
-              val newPens = p.pens.map {
-                pen =>
-                  val newPenName = penRenames.getOrElse(pen.display, pen.display)
-                  pen.copy(display = newPenName)
-              }
-              p.copy(display = newDisplay, pens = newPens)
-            case w => w
-          }
-          model.copy(widgets = updatedWidgets)
+   private def convertPlotAndPenNames( // you need to create a pipeline
+        model:       Model,               // Seq[Seq[String]] -> Seq[(String, String)] -> Complete
+        plotRenames: Map[String, String], // Apply each set of changes to its corresponding list
+        penRenames:  Map[String, Seq[(String, String)]]): Model = { // I need penRenames with list of list
+                                                     // List[Map[String,String]], possibly
+                                                     // this would need to be generated as we go
```

Comment:
I think the comments on this method should be moved before it into a more readable paragraph.  Also, sign and date.